src/fetchers/ethers/mooncats.ts

```diff
@@ -1,48 +1,46 @@
 import type { Address, NftMetadata } from "../../types"
 import type { EthersFetcherOptions } from "./types"
 
-import mooncat from "mooncatparser"
-import { addressesEqual } from "../../utils"
+import { addressesEqual, fetchImage, ipfsUrl } from "../../utils"
 
 const MOONCATS_WRAPPED_CONTRACT = "0x7c40c393dc0f283f318791d746d894ddd3693572"
 
 const MOONCATS_WRAPPED_ABI = [
   "function _tokenIDToCatID(uint256 tokenId) view returns (bytes5 catId)",
 ]
 
-const CANVAS_SIZE = 10
+// See https://www.reddit.com/r/MoonCatRescue/comments/m5d7mx/svg_imagery_of_all_rescued_mooncats/
+const MOONCATS_IPFS_CID =
+  "bafybeidk4zunuq56w2pf2sncexohlyqae62dzplljkbwswa7jwywh2dava"
 
-function imageUrl(catId: string): string | null {
-  const data = mooncat(catId)
+async function imageUrl(catId: string): Promise<string | null> {
+  const dir = catId.slice(4, 6)
+  const url = ipfsUrl(`ipfs://ipfs/${MOONCATS_IPFS_CID}/${dir}/${catId}.png`)
 
-  const width = CANVAS_SIZE * data.length
-  const height = CANVAS_SIZE * data[0].length
+  const image = await fetchImage(url)
+  const width = image.naturalWidth * 4
+  const height = image.naturalHeight * 4
+  const hPadding = width * 0.125
+  const vPadding = height * 0.125
 
   const canvas = document.createElement("canvas")
   canvas.width = width
   canvas.height = height
 
-  const ctx = canvas.getContext("2d", {
-    antialias: false,
-  }) as CanvasRenderingContext2D
+  const ctx = canvas.getContext("2d")
 
   if (ctx === null) {
     return null
   }
 
-  data.forEach((cells: string[], col: number) => {
-    cells.forEach((color: string, row: number) => {
-      if (color) {
-        ctx.fillStyle = color
-        ctx.fillRect(
-          Math.floor((col * CANVAS_SIZE) / 2 + width / 4),
-          Math.floor((row * CANVAS_SIZE) / 2 + height / 4),
-          Math.floor(CANVAS_SIZE / 2),
-          Math.floor(CANVAS_SIZE / 2)
-        )
-      }
-    })
-  })
+  ctx.imageSmoothingEnabled = false
+  ctx.drawImage(
```

Comment:
You're grabbing a PNG file from a remote URL, and then painting it onto a Canvas element? Why not simply use the remote URL itself as the URL to make an IMG tag of, rather than needing to re-paint every one?

Reply:
I do! The issue is that the images don’t have any padding, which makes them look bad when displayed directly. And since I was already doing this when the image was generated in canvas, I decided to keep the extra padding so it looks good :-)

Reply:
All right; that could also be accomplished with CSS, rather than spending the processing time to bake it into the image itself, but it's probably not a huge CPU hit to do this sort of processing.

Reply:
Exactly, it’s a small canvas that’s doesn’t get displayed on screen or inserted in the DOM tree, it shouldn’t have any noticeable cost. Otherwise CSS would certainly be an option yes in other contexts, but this library only returns a data structure, so the image has to be a URL.

Now I probably need to add my thinking as comments in the code itself, thanks for making me realize it :smile:

Reply:
I just realized you were the person that generated the assets and hosted them on IPFS, so thank you again :hugs: .

I was wondering if adding some padding on the .png directly could be a possibility? Or if you think there could be an issue doing that? That would make the process much more straightforward for useNft() haha

Reply:
You're welcome! The reason why I did _not_ add padding to the PNGs that I created is that because they're intended for re-use, I didn't want to presume how much padding was "enough" for all use-cases. All websites that want to include them can pick how much padding they need for their use-case. Having no padding also keeps the file size down (adding blank pixels adds a hair more to the file size per picture. It's not a lot, but in sites that may want to show whole galleries of lots of MoonCats, it could add up to larger network transfers to get them all). So, no, I'm not planning on adding in a hard-coded border/padding around the images at this time, to keep them as low file-size and as versatile as possible.

Reply:
Completely understandable. Thanks for having taken the time to give me a detailled answer!